regparser/history/annual.py

```diff
@@ -20,6 +21,17 @@
                 "CFR-{year}-title{title}-vol{volume}/xml/"
                 "CFR-{year}-title{title}-vol{volume}-part{part}.xml")
 
+# Matches any of the following:
+#    Parts 200 to 219
+#    Parts 200 to end
+#    Part 52 (§§ 52.1019 to 52.2019)
+PART_SPAN_REGEX = re.compile(
+    r'.*parts? ('
+    r'(?P<span>(?P<start>\d+) to ((?P<end>\d+)|(?P<end_literal>end)).*)'
+    r'|((?P<single_part>\d+) \(.*\))'
+    r')',
```

Comment:
Do we need the outermost parens (lines 29 and 32)? Might read ever so slightly better without them.

Also, would it be worthwhile to convert this to pyparsing? I find parser combinators to be much easier to reason about than regexes.